helpers/installcredprovider.ps1

```diff
@@ -17,7 +17,9 @@ param(
     # install the .NET 6 cred provider instead of NetCore3.1
     [switch]$InstallNet6 = $true,
     # install the .NET 8 cred provider instead of NetCore3.1
-    [switch]$InstallNet8
+    [switch]$InstallNet8,
+    # install the self-contained cred provider for the specified platform.
+    [string]$Platform
```

Comment:
platform here, but using rid in the Linux/Mac case? Is this going to only be win, linux, osx, or is this indeed the rid? win-x64, osx-arm64, etc.? if the later would go with RuntimeIdentifier here